common/aws.go

```diff
@@ -12,6 +12,26 @@ import (
 	"github.com/aws/aws-sdk-go-v2/service/sts"
 )
 
+// Generic paginator for AWS SDK v2
+type pageFetcherFunc[T any] func(nextToken *string) (items []T, next *string, err error)
+
+func Paginate[T any](fetch pageFetcherFunc[T]) ([]T, error) {
+	var allItems []T
+	var nextToken *string
+	for {
+		items, next, err := fetch(nextToken)
+		if err != nil {
+			return nil, err
+		}
+		allItems = append(allItems, items...)
```

Comment:
I suppose if this is to be a generic paginator, there are going to be scenarios where the complete list is excessively long.  This is really a question for a separate PR but I imagine we might want to have a stream of items or possibly be able to pass a mapper in to traverse the complete dataset and transform them either an item at a time or a batch at a time.